### DISCUSSION TITLE
Global Parameters

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#26121] [Q&A General]
All,
In order to switch DirichletBC to NeumannBC at t = 100 [sec], I am using the [Controls] block (please see [BCs] and [Controls] below).  Also, in a class in the [Materials] block, I need switch vis = 1e-4 to vis = 5e-4 at t = 100 [sec] (please see CalculateMaterial.C below).
It is nice if I can set a global parameter, e.g., t_switch = 100, and use it for both in the [Controls] block and CalculateMaterial.C as shown below (also, I can change the value for t_switch in the input file).

Please let me know whether [GlobalParams] in MOOSE does it and please help me where I can find examples to use it.
Please help. S. Thomas
```
[BCs]
    [./TT_ DirichletBC_BC]
    type = DirichletBC
    boundary = 'outer_boundary'
    variable = TT
    value = 0.0
   enable = false
  [../]
  [./TT_ NeumannBC _BC]
    type = NeumannBC
    boundary = = 'outer_boundary'
    variable = TTemp
    value = 0.0
   enable = false
  [../]
[ ]

[Controls]
   [./TT_DirchletBC_cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ DirichletBC_BC’
    start_time = 0.0
    end_time = 100.0
  [../]
   [./TT_= NeumannBC _cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ NeumannBC _BC’
    start_time = 100.0
    end_time = 1000.0
  [../]
[ ]

[Materials]
  [./mat_CalculateMaterial]
    type = CalculateMaterial
    variable = U
    block = '1 '
  [../]
[ ]
```
CalculateMaterial.C
```
If ( _t < 100.0 )
{ vis = 1e-4 }
else
{ vis = 5e-4}

```

```
[Controls]
   [./TT_DirchletBC_cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ DirichletBC_BC’
    start_time = 0.0
    end_time = t_switch
  [../]
   [./TT_= NeumannBC _cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ NeumannBC _BC’
    start_time = t_switch
    end_time = 1000.0
  [../]
```

CalculateMaterial.C

```
If ( _t < t_switch )
{ vis = 1e-4 }
else
{ vis = 5e-4}
```

**Comment by GiudGiud** [comment]
Can you do this, with input parameter substitution?
```
t_switch = 100
[Controls]
   [./TT_DirchletBC_cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ DirichletBC_BC’
    start_time = 0.0
    end_time = ${t_switch}
  [../]
   [./TT_= NeumannBC _cont]
    type = TimePeriod
    enable_objects = 'BCs::TT_ NeumannBC _BC’
    start_time = ${t_switch}
    end_time = 1000.0
  [../]
[ ]
```

**Reply by styyokuda** [reply]
Thank you, Guillaume.

Do you think this can also be applied for t_switch in the class of CalculateMaterial.C where
If ( _t < t_switch )
{ vis = 1e-4 }
else
{ vis = 5e-4}

Please let me know.
S. Thomas

**Reply by GiudGiud** [reply]
Absolutely 

Just make it to be a parameter (Real for the type) and set it from the input file
